gtest/src/log.rs

```diff
@@ -18,8 +21,40 @@ impl CoreLog {
             dest: other.dest,
             payload: other.payload.into_raw(),
             exit_code: other.reply.map(|(_, code)| Some(code)).unwrap_or_default(),
+            message_id: other.id
         }
     }
+
+    pub(crate) fn generate_reply(&self, payload: Payload, message_id: MessageId, value: u128) -> Message{
+        Message{
+            source: self.source,
+            dest: self.dest,
+            payload,
+            gas_limit: None,
+            value,
+            id: message_id,
+            reply: match self.exit_code {
+                None => {None}
+                Some(exit_code) => {Some((self.message_id, exit_code))}
+            }
+        }
+    }
+
+    pub(crate) fn reply(&self, system: &System, payload: Payload, value: u128) -> Result<RunResult, String>{
+        let message = self.generate_reply(
+            payload,
+            MessageId::from(system.fetch_inc_message_nonce()),
+            value
+        );
+        match system.0.borrow_mut().remove_message(&message.source, &message.id){
+            Result::Ok(_) => {Result::Ok(system.send_message(message))},
+            Result::Err(e) => {Result::Err(e)}
+        }
```

Comment:
Use `Result::map()`:

```suggestion
        system.0.borrow_mut().remove_message(&message.source, &message.id)
            .map(|_| system.send_message(message))
```